### PR TITLE
[Core] Fix pg stats not imported

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently, pg stats are not imported because it didn't use the enum "HISTOGRAM/GAUGE", but "Histogram/Gauge".

Added a static assert check that could help improving this as @iycheng suggested

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_metrics_agent.py [hunk 1]
```diff
@@ -59,6 +59,12 @@
     "ray_operation_run_time_ms",
     "ray_operation_queue_time_ms",
     "ray_operation_active_count",
+    # "ray_placement_group_creation_latency_ms",
+    # "ray_placement_group_scheduling_latency_ms",
```
Comment (iycheng) [inline python/ray/tests/test_metrics_agent.py:63]: should we uncomment them?

Comment (rkooo567) [reply python/ray/tests/test_metrics_agent.py:63]: This requires me to create a pg to export, so I commented it out... maybe I can create some pgs to make it work?

**Review by fishbone — APPROVED** [review]

**Comment by rkooo567** [comment]
Will merge after exporting stats for pgs

**Comment by rkooo567** [comment]
Java failure must be unrelated (it seems to be a flaky test)

**Pull request merged**